apps/events-helsinki/src/domain/event/__tests__/EventPageContainer.test.tsx

```diff
@@ -222,40 +222,91 @@ it("should show error info when event doesn't exist", async () => {
   });
 });
 
-describe(`SIMILAR_EVENTS feature flag`, () => {
-  it('shows similar events when flag is on', async () => {
-    advanceTo('2020-10-01');
-    renderComponent({ event: event, loading: false, showSimilarEvents: true });
-    await waitFor(() => {
-      expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
-    });
+it('shows similar events when SIMILAR_EVENTS flag is on', async () => {
+  advanceTo('2020-10-01');
+  const eventNoKeywords = { ...event };
+  eventNoKeywords.keywords[0].id = 'yso:p916';
+  renderComponent(
+    { event: eventNoKeywords, loading: false, showSimilarEvents: true },
+    [
+      ...mocks,
+      createEventListRequestAndResultMocks({
+        variables: {
+          allOngoing: true,
+          internetBased: undefined,
+          keywordOrSet2: ['yso:p916'],
+          language: undefined,
+          pageSize: 100,
+          publisherAncestor: null,
+          eventType: [EventTypeId.General],
+        },
+        response: similarEvents,
+      }),
+    ]
+  );
+  await waitFor(() => {
+    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+  });
+  await waitForLoadingCompleted();
+
+  await waitFor(() => {
     expect(
       screen.getByRole('heading', {
         name: translations.event.similarEvents.title,
       })
     ).toBeInTheDocument();
+  });
```

Comment:
No need for change, but findByRole always expects to find something, but still awaits it to appear. So this is kind of equal but simpler.

```suggestion
     await screen.findByRole('heading', {
        name: translations.event.similarEvents.title,
      })
```

Also the waiting for loading-spinners to be gone should be needless, because we are waiting for similar events heading to appear anyway.